tlsn/benches/bin/prover.rs

```diff
@@ -0,0 +1,183 @@
+use std::{
+    io::Write,
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc,
+    },
+    time::Instant,
+};
+
+use anyhow::Context;
+use futures::{AsyncReadExt, AsyncWriteExt};
+use tlsn_benches::{
+    config::{BenchInstance, Config},
+    metrics::Metrics,
+    set_interface, PROVER_INTERFACE,
+};
+
+use tlsn_core::Direction;
+use tlsn_server_fixture::{CA_CERT_DER, SERVER_DOMAIN};
+use tokio::io::{AsyncRead, AsyncWrite};
+use tokio_util::{
+    compat::TokioAsyncReadCompatExt,
+    io::{InspectReader, InspectWriter},
+};
+
+use tlsn_prover::tls::{Prover, ProverConfig};
+use tracing_subscriber::{fmt::format::FmtSpan, EnvFilter};
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    let config_path = std::env::var("CFG").unwrap_or_else(|_| "bench.toml".to_string());
+    let config: Config = toml::from_str(
+        &std::fs::read_to_string(config_path).context("failed to read config file")?,
+    )
+    .context("failed to parse config")?;
+
+    tracing_subscriber::fmt()
+        .with_env_filter(EnvFilter::from_default_env())
+        .with_span_events(FmtSpan::NEW | FmtSpan::CLOSE)
+        .init();
+
+    let ip = std::env::var("VERIFIER_IP").unwrap_or_else(|_| "10.10.1.1".to_string());
+    let port: u16 = std::env::var("VERIFIER_PORT")
+        .map(|port| port.parse().expect("port is valid u16"))
+        .unwrap_or(8000);
+    let verifier_host = (ip.as_str(), port);
+
+    let mut file = std::fs::OpenOptions::new()
+        .create(true)
+        .append(true)
+        .open("metrics.csv")
+        .context("failed to open metrics file")?;
+    let mut metric_wtr = csv::Writer::from_writer(&mut file);
+
+    for bench in config.benches {
+        let instances = bench.flatten();
+        for instance in instances {
+            println!("{:?}", &instance);
+
+            let io = tokio::net::TcpStream::connect(verifier_host)
+                .await
+                .context("failed to open tcp connection")?;
+            metric_wtr.serialize(
+                run_instance(instance, io)
+                    .await
+                    .context("failed to run instance")?,
+            )?;
+            metric_wtr.flush()?;
+        }
+    }
+
+    drop(metric_wtr);
```

Comment:
why do `metric_wtr` and `file` need to be manually dropped here? instead of automatically at the end of `main()`